pkg/controller/servicediscovery/privatednsnamespace/hooks.go

```diff
@@ -0,0 +1,160 @@
+package privatednsnamespace
+
+import (
+	"context"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	cpresource "github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/servicediscovery/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+func (e *external) observeImpl(ctx context.Context, cpm cpresource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	ret := managed.ExternalObservation{}
+	cr, ok := cpm.(*svcapitypes.PrivateDNSNamespace)
+	if !ok {
+		return ret, errors.New(errUnexpectedObject)
+	}
+	externalName := meta.GetExternalName(cr)
+	if externalName == "" {
+		return ret, nil
+	}
+
+	if cr.Status.AtProvider.OperationID == nil || *cr.Status.AtProvider.OperationID == "" {
+		return ret, nil
+	}
+	opInput := &svcsdk.GetOperationInput{
+		OperationId: cr.Status.AtProvider.OperationID,
+	}
+
+	opReqResp, err := e.client.GetOperationWithContext(ctx, opInput)
+	if err != nil {
+		return ret, errors.Wrap(err, "get-operation failed")
+	}
+
+	if *opReqResp.Operation.Status == "PENDING" || *opReqResp.Operation.Status == "SUBMITTED" {
```

Comment:
`opReqResp.Operation.Status` is a pointer and might be nil in some edge cases. This will cause a panic if you try to deref it.
Better to use `awsclient.StringValue`

Reply:
moved all string pointer dereferences into awsclient.StringValue calls